backend/src/main/java/com/now/naaga/gameresult/repository/GameResultRepository.java

```diff
@@ -3,8 +3,14 @@
 import com.now.naaga.gameresult.domain.GameResult;
 import java.util.List;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 
 public interface GameResultRepository extends JpaRepository<GameResult, Long> {
 
     List<GameResult> findByGameId(final Long gameId);
+
+    @Query("SELECT r FROM GameResult r JOIN FETCH r.game g LEFT JOIN FETCH g.hints Where g.player.id = :playerId")
```

Comment:
P5:
실행을 해봤는데
GameRecord에서 gameResult.getGame().getPlace().getPosition()라는 부분이 있더라고
이때, game-place간에도 n+1문제가 발생하고 place-player는 즉시 로딩돼서, 아래와같은 쿼리가 추가적으로 전송돼
<img width="321" alt="image" src="https://github.com/woowacourse-teams/2023-naaga/assets/100172683/63625a58-7fa1-4b48-86b7-a1939408924f">

해당 문제가 큰 문제라고 생각하지 않으면 넘어가도 좋지만, 
만약, 한번에 Game객체 내의 필요한 정보를 가져오는 것이 목적이라면, 
1. Place객체의 Player를 지연 로딩으로 변경하고 
2. game-place간의 n+1문제를 해결하기 위해 아래와 같은 쿼리를 사용하는 것도 좋을 것 같아.

```suggestion
        @Query("SELECT r FROM GameResult r JOIN FETCH r.game g LEFT JOIN FETCH g.hints JOIN FETCH g.place Where g.player.id = :playerId")
```

Reply:
이레 천재? GameRecord에서 목적지 정보를 사용하기때문에 쿼리가 한번 더 날아가는 거였구나...바로 수정하겠습니다!

Reply:
이레 천재? GameRecord에서 목적지 정보를 사용하기때문에 쿼리가 한번 더 날아가는 거였구나...바로 수정하겠습니다!

Reply:
또한 힌트가 존재하지 않을때 힌트를 출력해보니 null이 아닌 빈 리스트를 갖게되는데요..! 이래생각은 어떠신가요!?

Reply:
pr에 남겨주신 글에 대한 답변은..! 이레 말씀처럼 게임 결과를 조회하는 모든 부분에 현재 수정이 필요한 것이 맞습니다. 하지만 이 부분은 현재 통계조회를 수정하는 pr 범위에 벗어난다고 생각하여 수정하지 않았습니다!

Reply:
> 또한 힌트가 존재하지 않을때 힌트를 출력해보니 null이 아닌 빈 리스트를 갖게되는데요..! 이래생각은 어떠신가요!?
Game객체에서 힌트를 출력할 때는 빈 리스트를 갖는 게 맞지 않을까요?
사실 질문의 내용을 정확히 파악하지 못했숩니다!

> pr에 남겨주신 글에 대한 답변은..! 이레 말씀처럼 게임 결과를 조회하는 모든 부분에 현재 수정이 필요한 것이 맞습니다. 하지만 이 부분은 현재 통계조회를 수정하는 pr 범위에 벗어난다고 생각하여 수정하지 않았습니다!
넵 제가 해당 내용 슬랙에 공유하고 빠른시일 내에 이슈 파도록 하겠습니다